pkg/events/parse_args_helpers.go

```diff
@@ -68,13 +68,14 @@ func parseMemProtAlert(arg *trace.Argument, alert uint32) {
 }
 
 func parseSyscall(arg *trace.Argument, id int32) {
-	if Core.IsDefined(ID(id)) {
-		eventDefinition := Core.GetDefinitionByID(ID(id))
-		if eventDefinition.IsSyscall() {
-			arg.Value = eventDefinition.GetName()
-			arg.Type = "string"
-		}
+	// bypass the lock contention accessing the read-only map directly
+	def, ok := CoreEvents[ID(id)]
```

Comment:
Is there any issue of doing that, accessing directly and no using through the Core API?

Reply:
Let me double check this.

Reply:
Well, good catch @rscampos. Currently this change isn't a problem, but in the future it will cause data races due to the upcoming runtime changes to `CoreEvents`. That made me wonder why keep the non-changing definitions (syscalls) under lock contention? Would it be feasible to segregate them from the sigs? @yanivagman @NDStrahilevitz WDYT?

Reply:
Well, this brings a significant improvement in CPU time, although it will cause some concurrency issues in the future. I'm in favor of merging it (giving more information in the comment) to be a starting point for discussion about internal event segregation. If that's not feasible, we can easily fall back to the lock-contention way of retrieving the definition.